go-controller/pkg/allocator/pod/pod_annotation.go

```diff
@@ -265,6 +275,25 @@ func allocatePodAnnotationWithRollback(
 		IPs:      podAnnotation.IPs,
 		MAC:      podAnnotation.MAC,
 		TunnelID: podAnnotation.TunnelID,
+		Primary:  false,
+	}
+	if util.IsNetworkSegmentationSupportEnabled() {
+		tentative.Primary = netInfo.IsSecondary() && netInfo.IsPrimaryNetwork()
+		if !netInfo.IsSecondary() { // only do this extra bit if needed
+			var podNamespace *v1.Namespace
+			podNamespace, err = namespaceLister.Get(pod.Namespace)
+			if err != nil {
+				return
+			}
+			activeNetworkForPod := util.GetNamespaceActiveNetwork(podNamespace)
+			if activeNetworkForPod == nadName {
+				tentative.Primary = true
+			}
+		}
+	} else if !netInfo.IsSecondary() {
+		// if user defined network segmentation is not enabled
+		// then we know pod's primary network is "default"
+		tentative.Primary = true
```

Comment:
check if pods on upgrade should have this new field reflected in them? generally speaking any pod annotation with no changes will not have this reflected.

Reply:
upgrades should be tested for this. perhaps even pod annotations needs to be behind feature gate if we were to start using this field in 4.16 etc? UGH.

Reply:
hmm maybe its just better to put the pod annotation also behind a feature gate...